traversal/scanner/GraphIterator.java

```diff
@@ -115,6 +117,11 @@ else if (state == State.INIT) {
         }
     }
 
+    private void setCompleted() {
+        state = State.COMPLETED;
+        recycle();
```

Comment:
it's possible to fail half way through a `GraphIterator` because something in the middle of the plan does not have an answer. However, this means we have to recycle all the previous iterators that took us to this point!